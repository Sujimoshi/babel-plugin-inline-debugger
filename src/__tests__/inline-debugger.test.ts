import { runApp } from './fixtures/app';

describe('Inline Debugger', () => {
  beforeAll(() => {
    // Clear debug data before all tests
    global.inlineDebugger.clearData();
  });

  afterAll(() => {
    // Clean up after all tests
    global.inlineDebugger.clearData();
  });

  test('should monitor expressions in real files with imports', async () => {
    // Run the app with monitoring
    const result = await runApp();
    
    // Get debug data
    const debugData = global.inlineDebugger.getData();
    
    // Convert absolute paths to relative paths for consistent snapshots
    const normalizedData = debugData.map(item => ({
      ...item,
      filePath: item.filePath.startsWith(process.cwd()) 
        ? item.filePath.replace(process.cwd() + '/', '')
        : item.filePath
    }));
    
    // Snapshot the debug data
    expect(normalizedData).toMatchSnapshot();
    
    // Verify the result structure
    expect(result).toHaveProperty('math');
    expect(result).toHaveProperty('geometry');
    expect(result).toHaveProperty('user');
    expect(result).toHaveProperty('error');
    
    // Verify math operations
    expect(result.math).toEqual({
      sum: 8,
      product: 15,
      powerResult: 125
    });
    
    // Verify geometry calculations
    expect(result.geometry).toEqual({
      area: 50,
      perimeter: 30
    });
    
    // Verify user operations
    expect(result.user).toEqual({
      formatted: 'John (25)',
      isValid: true
    });
    
    // Verify error handling
    expect(result.error).toBe('Name is required');
  });

  test('should track file paths correctly', async () => {
    // Run the app to generate debug data
    await runApp();
    
    const debugData = global.inlineDebugger.getData();
    
    // Check that file paths are included
    const filePaths = debugData.map(item => item.filePath);
    expect(filePaths.length).toBeGreaterThan(0);
    expect(filePaths.every(path => typeof path === 'string')).toBe(true);
  });

  test('should only monitor lines with //? comments', async () => {
    // Run the app to generate debug data
    await runApp();
    
    const debugData = global.inlineDebugger.getData();
    
    // All monitored items should have line numbers (including 0 for some cases)
    debugData.forEach(item => {
      expect(item.line).toBeDefined();
      expect(item.line).toBeGreaterThanOrEqual(0);
    });
  });

  test('should handle different types of monitoring', async () => {
    // Run the app to generate debug data
    await runApp();
    
    const debugData = global.inlineDebugger.getData();
    
    // Check for different types
    const types = [...new Set(debugData.map(item => item.type))];
    expect(types.length).toBeGreaterThan(0);
    expect(types.every(type => typeof type === 'string')).toBe(true);
  });
});
