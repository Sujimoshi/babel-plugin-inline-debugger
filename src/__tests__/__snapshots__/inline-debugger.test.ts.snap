// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline Debugger should monitor expressions in real files with imports 1`] = `
[
  {
    "called": "5",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 27,
    "type": "variable",
    "variable": "a",
  },
  {
    "called": "3",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 28,
    "type": "variable",
    "variable": "b",
  },
  {
    "called": "8",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 6,
    "type": "return",
  },
  {
    "called": "8",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 30,
    "type": "variable",
    "variable": "sum",
  },
  {
    "called": "8",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 6,
    "type": "return",
  },
  {
    "called": "15",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 10,
    "type": "return",
  },
  {
    "called": "15",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 31,
    "type": "variable",
    "variable": "product",
  },
  {
    "called": "15",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 10,
    "type": "return",
  },
  {
    "called": "125",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 14,
    "type": "return",
  },
  {
    "called": "125",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 32,
    "type": "variable",
    "variable": "powerResult",
  },
  {
    "called": "125",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 14,
    "type": "return",
  },
  {
    "called": [
      ""Math results:"",
      "{"sum":8,"product":15,"powerResult":125}",
    ],
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 34,
    "type": "log",
  },
  {
    "called": "10",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 37,
    "type": "variable",
    "variable": "length",
  },
  {
    "called": "5",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 38,
    "type": "variable",
    "variable": "width",
  },
  {
    "called": "50",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 18,
    "type": "return",
  },
  {
    "called": "50",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 40,
    "type": "variable",
    "variable": "area",
  },
  {
    "called": "50",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 18,
    "type": "return",
  },
  {
    "called": "30",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 22,
    "type": "return",
  },
  {
    "called": "30",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 41,
    "type": "variable",
    "variable": "perimeter",
  },
  {
    "called": "30",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/math.ts",
    "line": 22,
    "type": "return",
  },
  {
    "called": [
      ""Area calculations:"",
      "{"area":50,"perimeter":30}",
    ],
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 43,
    "type": "log",
  },
  {
    "called": "{"name":"John","age":25}",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 46,
    "type": "variable",
    "variable": "user",
  },
  {
    "called": ""John (25)"",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/user.ts",
    "line": 11,
    "type": "return",
  },
  {
    "called": ""John (25)"",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 47,
    "type": "variable",
    "variable": "formatted",
  },
  {
    "called": ""John (25)"",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/user.ts",
    "line": 11,
    "type": "return",
  },
  {
    "called": "true",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/user.ts",
    "line": 15,
    "type": "return",
  },
  {
    "called": "true",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 48,
    "type": "variable",
    "variable": "isValid",
  },
  {
    "called": "true",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/user.ts",
    "line": 15,
    "type": "return",
  },
  {
    "called": [
      ""User operations:"",
      "{"formatted":"John (25)","isValid":true}",
    ],
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 50,
    "type": "log",
  },
  {
    "called": """",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 53,
    "type": "variable",
    "variable": "errorMessage",
  },
  {
    "called": "{}",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/user.ts",
    "line": 20,
    "type": "throw",
  },
  {
    "called": "{}",
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/user.ts",
    "line": 0,
    "type": "throw",
  },
  {
    "called": [
      ""Caught error:"",
      ""Name is required"",
    ],
    "filePath": "/Users/isolomakha/Projects/inline-debugger/src/__tests__/fixtures/app.ts",
    "line": 58,
    "type": "log",
  },
]
`;
